### ISSUE TITLE
source activate borked due to PS1 prompt coloring

### ISSUE THREAD

**Issue by ijstokes** [conda/conda#4494]
TL;DR: my `.bash_profile` (included below) does `PS1` prompt coloring. Worked before, but now with Anaconda 4.3 `activate` borks part way through leaving me with a bad `PATH` and general brokenness.

I've checked as much as I can to confirm that both the `alias` operations and the `PS1` prompt coloring are working properly (including when I'm in a git repo and outside of one).  It all seems fine.  Until I use the Conda `activate` script.

Details (I only figured out the problem part way through writing this out):

Yesterday I did everything I could to annihilate every last piece of Anaconda on my laptop, removing directories, "dot" directories, "dot" files, etc. with abandon.  I then installed Anaconda 4.3 from scratch.  I've created one Conda environment called "rbokeh" that contains "r-essentials" and a bunch of other things.  I also clicked the "install R Studio" button in Navigator, so my root environment has had "r-essentials" (and other stuff) added to it.

Today when I tried to go back to my "rbokeh" environment I started up a new terminal session and invoked `source activate rbokeh`.  I got the following as output:

```
$ source activate rbokeh
Traceback (most recent call last):
  File "/Users/ijstokes/anaconda/bin/conda", line 6, in <module>
    sys.exit(conda.cli.main())
  File "/Users/ijstokes/anaconda/lib/python3.6/site-packages/conda/cli/main.py", line 154, in main
    activate.main()
  File "/Users/ijstokes/anaconda/lib/python3.6/site-packages/conda/cli/activate.py", line 135, in main
    activation_path = get_activate_path(shelldict)
  File "/Users/ijstokes/anaconda/lib/python3.6/site-packages/conda/cli/activate.py", line 109, in get_activate_path
    binpath = binpath_from_arg(sys.argv[3], shelldict=shelldict)
  File "/Users/ijstokes/anaconda/lib/python3.6/site-packages/conda/cli/activate.py", line 66, in binpath_from_arg
    prefix = prefix_from_arg(arg, shelldict=shelldict)
  File "/Users/ijstokes/anaconda/lib/python3.6/site-packages/conda/cli/activate.py", line 58, in prefix_from_arg
    raise CondaValueError('Could not find environment: %s' % native_path)
conda.exceptions.CondaValueError: Value error: Could not find environment: /Users/ijstokes/anaconda/envs/dspyr
-bash: sed: command not found
-bash: grep: command not found
-bash: grep: command not found
-bash: sed: command not found
```

I have no idea where it could be picking up `/Users/ijstokes/anaconda/envs/dspyr`.  That was deleted along with everything else as part of my pre-Anaconda-4.3-annihilation.

But something bad happened as part of the `activate` script running, and I don't know what.

After getting this discouraging output I tried to see if I had a weird `activate` script on my `$PATH`, so I tried `which activate`.  This put me on to the fact that my `$PATH` was borked:

```
(rbokeh) ijstokes@0194-ijstokes DataSciencePythonR $ which activate
-bash: which: command not found
-bash: grep: command not found
-bash: sed: command not found

(rbokeh) ijstokes@0194-ijstokes DataSciencePythonR $ echo $PATH
/Users/ijstokes/anaconda/envs/rbokeh/bin:
-bash: grep: command not found
-bash: sed: command not found
```

I don't have time now to investigate much further, but perhaps others have seen this.

FWIW, I looked into my `~/.bash*` files to see if I had some weird settings in there, and I noticed that Anaconda is added 4 times to my `$PATH` (this is my complete `.bash_profile`:

```
# added by Anaconda 2.2.0 installer
export PATH="/Users/ijstokes/anaconda/bin:$PATH"

alias ls="ls -G"
alias less="less -r"
alias grep="grep --color=auto"

function color_my_prompt {
    local __user_and_host="\[\033[01;32m\]\u@\h"
    local __cur_location="\[\033[01;34m\]\W"
    local __git_branch_color="\[\033[31m\]"
    #local __git_branch="\`ruby -e \"print (%x{git branch 2> /dev/null}.grep(/^\*/).first || '').gsub(/^\* (.+)$/, '(\1) ')\"\`"
    local __git_branch='`git branch 2> /dev/null | grep -e ^* | sed -E  s/^\\\\\*\ \(.+\)$/\(\\\\\1\)\ /`'
    local __prompt_tail="\[\033[35m\]$"
    local __last_color="\[\033[00m\]"
    export PS1="$__user_and_host $__cur_location $__git_branch_color$__git_branch$__prompt_tail$__last_color "
}
color_my_prompt

export CONDA_ENVS=~/anaconda/envs

# added by Anaconda3 4.2.0 installer
export PATH="/Users/ijstokes/anconda3new/anaconda/bin:$PATH"

# added by Anaconda3 4.2.0 installer
export PATH="/Users/ijstokes/anaconda/bin:$PATH"

# added by Anaconda3 4.2.0 installer
export PATH="/Users/ijstokes/anaconda/bin:$PATH"

# added by Anaconda3 4.3.0 installer
export PATH="/Users/ijstokes/anaconda/bin:$PATH"
```

Fresh terminal session:

```
$ echo $PATH
/Users/ijstokes/anaconda/bin:/Users/ijstokes/anaconda/bin:/Users/ijstokes/anaconda/bin:/Users/ijstokes/anconda3new/anaconda/bin:/Users/ijstokes/anaconda/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

Multiple `anaconda/bin` entries do not seem relevant: I've now removed them and the problem persists.

I wondered if it might be my `alias` operations, but no dice: commented these out and problem continuued.

**Comment by ijstokes** [comment]
In case it is helpful:

```
$ conda info -a
Current conda install:

               platform : osx-64
          conda version : 4.3.8
       conda is private : False
      conda-env version : 4.3.8
    conda-build version : not installed
         python version : 3.6.0.final.0
       requests version : 2.12.4
       root environment : /Users/ijstokes/anaconda  (writable)
    default environment : /Users/ijstokes/anaconda
       envs directories : /Users/ijstokes/anaconda/envs
          package cache : /Users/ijstokes/anaconda/pkgs
           channel URLs : https://conda.anaconda.org/anaconda-fusion/osx-64
                          https://conda.anaconda.org/anaconda-fusion/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/osx-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /Users/ijstokes/.condarc
           offline mode : False
             user-agent : conda/4.3.8 requests/2.12.4 CPython/3.6.0 Darwin/16.4.0 OSX/10.12.3
                UID:GID : 502:20

# conda environments:
#
rbokeh                   /Users/ijstokes/anaconda/envs/rbokeh
root                  *  /Users/ijstokes/anaconda

sys.version: 3.6.0 |Anaconda 4.3.0 (x86_64)| (default...
sys.prefix: /Users/ijstokes/anaconda
sys.executable: /Users/ijstokes/anaconda/bin/python
conda location: /Users/ijstokes/anaconda/lib/python3.6/site-packages/conda
conda-build: None
conda-env: /Users/ijstokes/anaconda/bin/conda-env
conda-server: /Users/ijstokes/anaconda/bin/conda-server
user site dirs: 

CIO_TEST: <not set>
CONDA_DEFAULT_ENV: <not set>
CONDA_ENVS_PATH: <not set>
DYLD_LIBRARY_PATH: <not set>
PATH: /Users/ijstokes/anaconda/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
PYTHONHOME: <not set>
PYTHONPATH: <not set>

License directories:
    /Users/ijstokes/.continuum
    /Users/ijstokes/Library/Application Support/Anaconda
    /Users/ijstokes/anaconda/licenses
License files (license*.txt):
Package/feature end dates:
```

Conda/Anaconda versions:

```
$ conda list | grep -i conda
# packages in environment at /Users/ijstokes/anaconda:
anaconda                  4.3.0               np111py36_0  
anaconda-client           1.6.0                    py36_0  
anaconda-navigator        1.4.3                    py36_0  
conda                     4.3.8                    py36_0  
conda-env                 2.6.0                         0  
```

And:

```
ijstokes@0194-ijstokes zeppelin (master) $ ls -Fla ~/anaconda/envs/
total 0
drwxr-xr-x   3 ijstokes  staff  102 Jan 31 20:23 ./
drwxr-xr-x  24 ijstokes  staff  816 Jan 31 20:21 ../
drwxr-xr-x  22 ijstokes  staff  748 Jan 31 20:36 rbokeh/
```

**Comment by csoja** [comment]
@kalefranz @mingwandroid

**Comment by mingwandroid** [comment]
This is in `conda`, I don't think `rbokeh` is a pertinent detail?

**Comment by ijstokes** [comment]
@mingwandroid `rbokeh` should have nothing to do with it.  It just happens to be the only Conda environment I have on my system right now, so the only thing I had to `activate`.

I'm still totally stumped by the reference to `dspyr` -- a Conda environment from before I blew away my Anaconda 4.2 installation.

**Comment by mingwandroid** [comment]
@ijstokes did you close your terminal after blowing everything away? I'm thinking some env. var may have persisted.

**Comment by ijstokes** [comment]
@mingwandroid  I did start a completely fresh terminal window, but I did not quit Terminal.app and restart it.  I did not see (though you can check above) any environment variables that would have impacted Conda.  I mean I tried many things to fix the problem and started many fresh terminal sessions while editing my `.bash_profile` before finally discovering that commenting out the `color_my_prompt` bash function made things work again (and un-commenting it made things break again).

Now that I've done that (restarted Terminal.app completely) things seem to work! I'm closing this issue. But that is weird!

Previously I had experienced CMD-N as completely fresh login sessions (fresh environment variables), whereas CMD-T (new tab) would open a new shell that inherited environment variables.  You can see that `conda info` wasn't showing any obvious environment variables that would cause things to behave weirdly.

**Comment by mingwandroid** [comment]
Glad you got it working. macOS env. vars are strange beasts! @goanpeca and I had some fun with one called `QT_API`, we couldn't for the life of us figure out where it was set, what did you do with that one in the end Gonzalo?

**Comment by mingwandroid** [comment]
@ijstokes, I wonder if `CONDA_PREFIX` was set? It's not in the list though.
